types/src/database_sql.rs

```diff
@@ -398,7 +396,76 @@ impl Database {
             "queries/update_word.sql",
             word.id,
             &source as _,
-            &commentary as _
+            &commentary as _,
+        )
+        .execute(&self.client)
```

Comment:
Oh all of these queries inside this function need to go into a transaction so that if one fails the whole thing gets rolled back. Make sure to commit or rollback the transaction at any and all return points (since you have an early return in here).